File: src/main/java/guru/xgm/language/angelscript/lang/folding/AngelScriptCodeFoldingSettings.java
package guru.xgm.language.angelscript.lang.folding;

import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.SettingsCategory;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import lombok.Getter;
import lombok.Setter;
import org.jetbrains.annotations.NotNull;

@Setter
@Getter
@State(name = "AngelScriptCodeFoldingSettings", storages = @Storage("editor.xml"), category = SettingsCategory.CODE)
public class AngelScriptCodeFoldingSettings implements PersistentStateComponent<AngelScriptCodeFoldingSettings> {
    private boolean foldEnum = false;

    public static AngelScriptCodeFoldingSettings getInstance() {
        return ApplicationManager.getApplication().getService(AngelScriptCodeFoldingSettings.class);
    }

    @Override
    public AngelScriptCodeFoldingSettings getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull final AngelScriptCodeFoldingSettings state) {
        XmlSerializerUtil.copyBean(state, this);
    }

}

File: src/main/kotlin/raft/war/language/angelscript/folding/AngelScriptCodeFoldingSettings.kt
package raft.war.language.angelscript.folding

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.SettingsCategory
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

@State(name = "AngelScriptCodeFoldingSettings", storages = [Storage("editor.xml")], category = SettingsCategory.CODE)
class AngelScriptCodeFoldingSettings : PersistentStateComponent<AngelScriptCodeFoldingSettings> {
    var isFoldEnum: Boolean = false

    override fun getState(): AngelScriptCodeFoldingSettings {
        return this
    }

    override fun loadState(state: AngelScriptCodeFoldingSettings) {
        XmlSerializerUtil.copyBean(state, this)
    }

    companion object {
        @JvmStatic
        val instance: AngelScriptCodeFoldingSettings
            get() = ApplicationManager.getApplication().getService(
                AngelScriptCodeFoldingSettings::class.java
            )
    }
}
